force-app/main/default/classes/FaxService.cls

```diff
@@ -164,12 +164,29 @@ public with sharing class FaxService implements Queueable, Database.AllowsCallou
     
     private static List<Case> queryCases(List<Id> caseIds) {
         return new List<Case>([
-            select Id, CaseNumber, status,Provider_Fax__c,Provider__r.name, Drug__r.Drug_Code__c 
+            select Id, CaseNumber, status,Provider_Fax__c,Provider__r.name,DEC__r.name, Drug__r.Drug_Code__c 
             from Case 
             where Id in :caseIds
         ]);
     }
 
+    private static string queryName(Id caseId) {
```

Comment:
@afreen1701 just take the case record so you don't have to make a trip to the database to query the case again.., eg.

```
public static void sendFax(Id caseId, Id templateId) {
   Case record = queryCases(new Id[] {caseId})[0];
   ...
   getRecipientName(record);
   ....
}

private static void runSendFax(List<Id> caseIds) {
   ...
   for (Id caseId : caseIds) {
      Case record = caseMap.get(caseId);
      ...
      getRecipientName(record);
   }
   ...
}
private static String getRecipientName(Case record) {
   if (record.Provider__r != null) return record.Provider__r.Name;
   else if (record.DEC__r != null return record.DEC__r.Name; 
   else return 'NO_RECIPIENT_NAME';
}
```